### DISCUSSION TITLE
Is it possible to make images expand on click?

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#2137] [Q&A] [answered]
I would like to utilize the attribution list expansion to display images alongside text.

The issue I face is, that the images are often too large in one dimention (either width which makes them not next to the text or height which may change things like headers)
I could obviously adjust the size to make it fit, but this would give the disadvantage that the images would become quite small which makes reading text on them harder.

I was now wondering if I can use certain attributes to change the size of the image and to also make them expand into their full resolution (Show in a glowbox or what it is called) when you click on them.

**Comment by squidfunk** [comment]
There's currently no out-of-the-box lightbox gallery solution, but there are many issues where people shared their solutions. Attributes (like size) can be changed using the attribute list extension, which can also be used to [align images](https://squidfunk.github.io/mkdocs-material/reference/images/#image-alignment). This extension can also be used to attach classnames to images to select them for displaying with a third party lightbox plugin.

**Reply by Andre601** [reply]
Do you have any solutions at hand?
I'm well aware about the attribution list extension and already explained my issue above about the alignment.

**Reply by squidfunk** [reply]
A quick search yields #1635. For more, I would also have to search the issue tracker, so no, I don't have a drop-in solution at-hand. Maybe somebody else can share his solution or you could search the issue tracker and share it, as I'm sure that in the past some were posted.

EDIT: I'm going to put an attribute-list based lightbox functionality on the roadmap for Insiders, as it was requested quite a few times.

**Comment by fveauvy** [comment, marked as answer]
I personally use [lightgallery-markdown](https://github.com/g-provost/lightgallery-markdown) Markdown extension.

See for yourself :

https://user-images.githubusercontent.com/38651917/102955813-06f04880-44d7-11eb-8a61-c6ee768c3c90.mov

**Reply by Andre601** [reply]
I assume this will go alongside the Material theme, or could there be complications?

**Reply by Andre601** [reply]
Also, to add to this am I not quite sure what name and/or version I should provide to get the extension itself, as the readme is incredibly vague about this particular part.

**Reply by fveauvy** [reply]
This extension works as intended with my use cases of the material theme.

I suggest you ask any questions regarding this extension to the dedicated repo.

**Comment by aparcar** [comment]
@squidfunk do you plan to integrate similar functionality directly to your mkdocs theme?

**Reply by squidfunk** [reply]
Currently no plans, but maybe some time in the future. The solution posted by @fveauvy already seems to work pretty well.

**Comment by aborruso** [comment]
@squidfunk in your site on click on the image, in example in [this page](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/?h=hi#navigation-tabs), you have the image opened.

How to have this behaviour?

Thank you

![image](https://user-images.githubusercontent.com/30607/138599927-03a2daa4-bfc2-4a10-b275-611370632daa.png)

**Reply by Andre601** [reply]
Did you actually check the response that is marked as an answer?

Also, consider checking https://github.com/g-provost/lightgallery-markdown/issues/8

**Reply by squidfunk** [reply]
It's just an image, that is linked to itself, e.g.:

``` markdown
[![navigation.tabs enabled]][navigation.tabs enabled]

  [navigation.tabs enabled]: ../assets/screenshots/navigation-tabs.png
```

**Reply by aborruso** [reply]
thank you @squidfunk

**Comment by aborruso** [comment]
@Andre601 is it the solution that @squidfunk use in his site?

Thank you

**Comment by copdips** [comment]
It could be done by [mkdocs-glightbox](https://github.com/blueswen/mkdocs-glightbox).

**Reply by keegoo** [reply]
already recommended by mkdocs-material itself in 2024.

https://squidfunk.github.io/mkdocs-material/reference/images/#lightbox
